src/renderer/i18n/index.js

```diff
@@ -0,0 +1,24 @@
+import i18n, { languageOptions, DEFAULT_LOCALE } from '../../locales'
+import { ipcRenderer } from 'electron'
+
+/**
+ * Fetch language config value from preference storage.
+ * @returns A promise carrying the language value.
+ */
+function asyncGetLanguage () {
```

Comment:
There is a global state that is passed to the renderer process at startup. It would be the best option to include the language entry there. It's called `global.marktext`, available in each file and parsed here:

https://github.com/marktext/marktext/blob/b43eda2586ae644c2571c90c225cfaf241a5f12e/src/renderer/bootstrap.js#L18

Reply:
Seems great, I've never discovered this entry before.

But implementing this for i18n support is a little bit complicated for me. Maybe I can pass langugae config via `_buildUrlWithSettings` (`src/main/windows/base.js:70`), then fetch it via `global.marktext`?

`asyncGetLanguage()` will be invoked in `src/renderer/pages/(app.vue|preferences.vue)`. Within this promise I set language.

See: `279ae61`.